openedx/core/djangoapps/user_api/views.py

```diff
@@ -921,10 +923,11 @@ def _apply_third_party_auth_overrides(self, request, form_desc):
                 if current_provider:
                     # Override username / email / full name
                     field_overrides = current_provider.get_register_form_data(
-                        running_pipeline.get('kwargs')
+                        running_pipeline.get('kwargs'),
+                        current_provider=current_provider
                     )
 
-                    for field_name in self.DEFAULT_FIELDS:
+                    for field_name in self.DEFAULT_FIELDS + self.EXTRA_TPA_FIELDS:
```

Comment:
I thought it'd be maybe better to separate out those fields that TPA wants specifically, in addition to the defaults. I contemplated just doing `self.DEFAULT_FIELDS + self.EXTRA_FIELDS` but I suppose it's less code maintenance to just have it check every field.